libs/feature/editor/src/lib/components/record-form/form-field/form-field-contacts-for-resource/form-field-contacts-for-resource.component.ts

```diff
@@ -0,0 +1,188 @@
+  standalone: true,
+  imports: [
+    DropdownSelectorComponent,
+    UiInputsModule,
+    CommonModule,
+    UiWidgetsModule,
+    AutocompleteComponent,
+    TranslateModule,
+    ContactCardComponent,
+  ],
+})
+export class FormFieldContactsForResourceComponent
+  implements OnInit, OnChanges
+{
+  @Input() control: FormControl<Individual[]>
+
+  allUser$: Observable<UserModel[]>
+
+  rolesToPick: string[] = [
+    'resource_provider',
+    'custodian',
+    'owner',
+    'point_of_contact',
+    'author',
+  ]
+
+  rolesToDisplay = []
+
+  allOrganizations: Map<string, Organization> = new Map()
+
+  constructor(
+    private platformServiceInterface: PlatformServiceInterface,
+    private organizationsServiceInterface: OrganizationsServiceInterface,
+    private changeDetectorRef: ChangeDetectorRef
+  ) {
+    this.allUser$ = this.platformServiceInterface.getUsers()
+  }
+
+  ngOnInit(): void {
+    this.organizationsServiceInterface.organisations$.subscribe(
+      (organizations) => {
+        this.allOrganizations = new Map<string, Organization>(
+          organizations.map((organization) => [organization.name, organization])
+        )
+        this.changeDetectorRef.markForCheck()
+      }
+    )
+  }
+
+  async ngOnChanges(changes: SimpleChanges): Promise<void> {
+    const contactsForResource = changes['control']
+
+    if (
+      contactsForResource &&
+      contactsForResource.currentValue !== contactsForResource.previousValue
+    ) {
+      const rolesToAdd = contactsForResource.currentValue.value.map(
+        (contact: Individual) => contact.role
+      )
+
+      rolesToAdd.forEach((role: string) => {
+        if (!this.rolesToDisplay.includes(role)) {
+          this.rolesToDisplay.push(role)
+        }
+      })
+
+      this.filterRolesToPick()
+
+      this.changeDetectorRef.markForCheck()
+    }
+  }
+
+  addRoleToDisplay(roleToAdd: string) {
+    this.rolesToDisplay.push(roleToAdd)
+    this.filterRolesToPick()
+  }
+
+  filterRolesToPick() {
+    this.rolesToPick = this.rolesToPick.filter(
+      (role) => !this.rolesToDisplay.includes(role)
+    )
+  }
+
+  removeContact(index: number) {
+    const newContactsforRessource = this.control.value.filter(
+      (_, i) => i !== index
+    )
+    this.control.setValue(newContactsforRessource)
+  }
+
+  protected roleToLabel(role: string): string {
+    return RoleLabels.get(role)
+  }
+
+  /**
+   * gn-ui-autocomplete
+   */
+  displayWithFn: (user: UserModel) => string = (user) =>
+    `${user.name} ${user.surname}`
+
+  /**
+   * gn-ui-autocomplete
+   */
+  autoCompleteAction = (query: string) => {
+    return this.allUser$.pipe(
+      switchMap((users) => [
+        users.filter((user) => user.username.includes(query)),
+      ]),
+      debounceTime(300),
+      distinctUntilChanged()
+    )
+  }
+
+  /**
+   * gn-ui-autocomplete
+   */
+  async addContact(contact: UserModel, role: string) {
+    let newContactsforRessource = {
```

Comment:
```suggestion
    let newContactsForRessource = {
```
really minor thing, sorry :D